aws/resource_aws_dlm_lifecycle_policy.go

```diff
@@ -0,0 +1,344 @@
+	result["resource_types"] = flattenStringList(policyDetails.ResourceTypes)
+	result["schedule"] = flattenDlmSchedules(policyDetails.Schedules)
+	result["target_tags"] = flattenDlmTags(policyDetails.TargetTags)
+
+	return []map[string]interface{}{result}
+}
+
+func expandDlmSchedules(cfg []interface{}) []*dlm.Schedule {
+	schedules := make([]*dlm.Schedule, len(cfg))
+	for i, c := range cfg {
+		schedule := &dlm.Schedule{}
+		m := c.(map[string]interface{})
+		if v, ok := m["create_rule"]; ok {
+			schedule.CreateRule = expandDlmCreateRule(v.([]interface{}))
+		}
+		if v, ok := m["name"]; ok {
+			schedule.Name = aws.String(v.(string))
+		}
+		if v, ok := m["retain_rule"]; ok {
+			schedule.RetainRule = expandDlmRetainRule(v.([]interface{}))
+		}
+		if v, ok := m["tags_to_add"]; ok {
+			schedule.TagsToAdd = expandDlmTags(v.(map[string]interface{}))
+		}
+		schedules[i] = schedule
+	}
+
+	return schedules
+}
+
+func flattenDlmSchedules(schedules []*dlm.Schedule) []map[string]interface{} {
+	result := make([]map[string]interface{}, len(schedules))
+	for i, s := range schedules {
+		m := make(map[string]interface{})
+		m["create_rule"] = flattenDlmCreateRule(s.CreateRule)
+		m["name"] = aws.StringValue(s.Name)
+		m["retain_rule"] = flattenDlmRetainRule(s.RetainRule)
+		m["tags_to_add"] = flattenDlmTags(s.TagsToAdd)
+		result[i] = m
+	}
+
+	return result
+}
+
+func expandDlmCreateRule(cfg []interface{}) *dlm.CreateRule {
+	c := cfg[0].(map[string]interface{})
+	createRule := &dlm.CreateRule{
+		Interval:     aws.Int64(int64(c["interval"].(int))),
+		IntervalUnit: aws.String(c["interval_unit"].(string)),
+	}
+	if v, ok := c["times"]; ok {
+		createRule.Times = expandStringList(v.([]interface{}))
+	}
+
+	return createRule
+}
+
+func flattenDlmCreateRule(createRule *dlm.CreateRule) []map[string]interface{} {
+	if createRule == nil {
+		return []map[string]interface{}{}
+	}
+
+	result := make(map[string]interface{})
+	result["interval"] = aws.Int64Value(createRule.Interval)
+	result["interval_unit"] = aws.StringValue(createRule.IntervalUnit)
+	result["times"] = flattenStringList(createRule.Times)
+
+	return []map[string]interface{}{result}
+}
+
+func expandDlmRetainRule(cfg []interface{}) *dlm.RetainRule {
+	return &dlm.RetainRule{
+		Count: aws.Int64(int64(cfg[0].(map[string]interface{})["count"].(int))),
+	}
+}
+
+func flattenDlmRetainRule(retainRule *dlm.RetainRule) []map[string]interface{} {
+	result := make(map[string]interface{})
+	result["count"] = aws.Int64Value(retainRule.Count)
+
+	return []map[string]interface{}{result}
+}
+
+func expandDlmTags(m map[string]interface{}) []*dlm.Tag {
+	var result []*dlm.Tag
+	for k, v := range m {
+		result = append(result, &dlm.Tag{
+			Key:   aws.String(k),
+			Value: aws.String(v.(string)),
+		})
+	}
+
+	return result
+}
+
+func flattenDlmTags(tags []*dlm.Tag) map[string]string {
+	result := make(map[string]string)
+	for _, t := range tags {
+		result[*t.Key] = *t.Value
```

Comment:
To prevent potential panics, we should prefer to use the SDK helpers in this case, e.g.

```go
result[aws.StringValue(t.Key)] = aws.StringValue(t.Value)
```